spell/rwa-checklists.md

```diff
@@ -0,0 +1,191 @@
+    * [ ] Global debt ceiling (`vat.Line`) is updated accordingly, UNLESS specifically instructed not to, via EITHER:
+        * `global` set to `true` in `increaseIlkDebtCeiling`/`decreaseIlkDebtCeiling`
+        * [`DssExecLib.setGlobalDebtCeiling(amount)`](https://github.com/makerdao/dss-exec-lib/blob/v0.0.9/src/DssExecLib.sol#L428)
+        * [`DssExecLib.increaseGlobalDebtCeiling(amount)`](https://github.com/makerdao/dss-exec-lib/blob/v0.0.9/src/DssExecLib.sol#L436)
+        * [`DssExecLib.decreaseGlobalDebtCeiling(amount)`](https://github.com/makerdao/dss-exec-lib/blob/v0.0.9/src/DssExecLib.sol#L445C14-L445C39)
+  * [ ] Liquidation oracle price is bumped via `RwaLiquidationOracleLike(MIP21_LIQUIDATION_ORACLE).bump(ilk, val)` pattern
+    * [ ] Comment above `bump` explains `val` computation via `// Note: the formula is: "debt_ceiling * [ (1 + rwa_stability_fee ) ^ (minimum_deal_duration_in_years) ] * liquidation_ratio"`
+    * [ ] Comment  above `bump` provides locally executable formula (e.g. `// bc -l <<< 'scale=18; 50000000 * e(l(1.07) * (3342/365)) * 1.00' | cast --to-wei`)
+      * [ ] The formula matches the example provided above
+      * [ ] `debt_ceiling` in the executable formula matches new debt ceiling set in the spell or the maximum possible debt ceiling in case of the enabled AutoLine
+      * [ ] `rwa_stability_fee` in the executable formula matches stability fee of the specified RWA found on chain
+      * [ ] `minimum_deal_duration_in_years` in the executable formula matches number found in the Exec Sheet of the spell containing relevant RWA onboarding
+      * [ ] `liquidation_ratio` in the executable formula matches liquidation ratio of the specified RWA found on chain
+      * [ ] Executing formula locally provides integer number that matches the `val` in the spell
+    * [ ] `val` makes sense in context of the [rate mechanism](https://github.com/makerdao/developerguides/blob/master/mcd/intro-rate-mechanism/intro-rate-mechanism.md)
+  * [ ] IF multiple RWA ilks are being combined into one, `val` calculation is done once per ilk and added to make the total, with separate executable formulas provided in comments. The existing `val` value can be retrieved by calling `read()` on `PIP_RWAXYZ` and converting the result into decimal
+  * [ ] Oracle price is updated via `DssExecLib.updateCollateralPrice(ilk)`
+  * IF soft liquidation explicitly requested to be triggered (via `.tell(ilk)`) AND debt ceiling (`vat.ilks(ilk).line`) is `0` (OR is being set to `0` in the current spell)
+      * [ ] `RwaLiquidationOracle.tell(ilk)` call is present
+      * [ ] IF `RWAXYZ_A_INPUT_CONDUIT` is an instance of [`TinlakeMgr`](https://github.com/centrifuge/tinlake-maker-lib/blob/master/src/mgr.sol) (it is a Centrifuge integration), additional `TinlakeMgr.tell()` call is present (in order to prevent further `TIN` redemptions in the Centrifuge pool)
```

Comment:
This has been in the [`spell-reviewer-mainnet-checklist`](https://github.com/makerdao/pe-checklists/blob/5208017c2c5e04f9ec03817856e06ff9d0fe9ad1/spell/spell-reviewer-mainnet-checklist.md?plain=1#L212) but I'm unsure it was followed in the previous spells.
(eg. [soft liquidation for RWA005-A ](https://github.com/makerdao/spells-mainnet/blob/af7bcbfa8fbfd39981f47808caeb3fbed30dc865/archive/2023-09-27-DssSpell/DssSpell.sol#L104-L110))

Could you guide me through where I can find related code or help me understand why this is needed?

Reply:
There's no official guide.
The Centrifuge Vaults are pretty much abandonware nowadays.
The TL;DR:
- Instead of the traditional input and output conduits, Centrifuge vaults have a component that belongs to them (even though they were written by people working for MakerDAO back then), dubbed [`TinlakeManager`](https://github.com/centrifuge/tinlake-maker-lib/blob/master/src/mgr.sol).
- The `TinlakeManager` instance associated with the vault fulfills both the role of the input and the output conduit.
- It also drives the interaction with the Centrifuge pools, so it can be seen as the adapter component that connects the Maker Protocol to the Centrifuge Protocol.
- It has some methods that are named the same as other RWA components, such as `wipe`, `lock`, `free`, `exit`, `draw` (from [`RwaUrn`](https://github.com/makerdao/rwa-toolkit/blob/master/src/urns/RwaUrn.sol)) and `tell`, `cull` (from [`RwaLiquidationOracle`](https://github.com/makerdao/rwa-toolkit/blob/master/src/oracles/RwaLiquidationOracle.sol)).
- `tell` and `cull` are **permissionless**, but such actions can only take place after the `RwaLiquidationOracle.tell()` and `RwaLiquidationOracle.cull()` have been called, respectively.
- It is important to call `tell` to prevent further redemptions in the related Centrifuge pool
  - It's not documented anywhere, but these calls were supposed to be handled by Centrifuge, however they haven't done it in the past and asked us to do it.
  - To prevent any unwanted behavior, we added this step to the checklist.
- As for `cull`, it's optional, as it only settles the state in `TinlakeManager`.
  - If an ilk was `cull`ed in `RwaLiquidationOracle`, technically there's nothing else that can be done through Centrifuge.

Reply:
Thank you for the explainer!